js/views/userPage/UserPageFollow.js

```diff
@@ -1,23 +1,104 @@
 
-export default class extends View {
+export default class extends BaseVw {
   constructor(options = {}) {
-    super({
-      className: 'userPageFollow',
-      events: {
-      },
-      ...options,
-    });
+    super(options);
+    this.options = options;
+    this.followType = options.followType;
+
+    if (!options.ownPage) {
+      this.followCol = new Follows(null, {
+        type: this.followType.toLowerCase(),
+        guid: this.model.id,
+      });
+      this.followCol.fetch().done(() => {
+        this[`updateViewer${this.followType}`](true);
+        this.render();
+      });
+
+      this.listenTo(app.ownFollowing, 'sync, update', () => {
+        this.updateViewerFollowers();
+        this.render();
+      });
+
+      this.listenTo(app.ownFollowers, 'update', () => {
+        this.updateViewerFollowing();
+        this.render();
+      });
+    } else {
+      this.followCol = app[`own${this.followType}`];
+      if (this.followType === 'Followers') {
+        this.followCol.fetch().done(() => {
+          this.render();
+        }); // TODO: pagination
+      } else {
+        this.render();
+      }
+
+      this.listenTo(app[`own${this.followType}`], 'sync, update', () => {
+        this.render();
+      });
+    }
   }
 
-  render() {
-    loadTemplate('userPage/userPageFollow.html', (t) => {
-      this.$el.html(t({
-        ...this.model.toJSON(),
-      }));
-    });
+  updateViewerFollowers() {
+    /* if the viewer follows/unfollows this user, add them to the followers list without a fetch */
+    if (this.followType === 'Followers' && !this.options.ownPage) {
+      // if the viewer has followed add them
+      if (followUtils.followedByYou(this.model.id)) {
+        this.followCol.add({ guid: app.profile.id });
+        // if the viewer has unfollowed remove them
+      } else if (!followUtils.followedByYou(this.model.id)) {
+        this.followCol.remove(app.profile.id); // remove by id
+      }
+    }
+  }
 
+  updateViewerFollowing(viaAPI = false) {
+    /* if the viewer is unfollowed or followed, update the following list */
+    if (this.followType === 'Following' && !this.options.ownPage) {
+      let followsYou;
+      if (viaAPI) {
+        // get via isFollowingMe API when it is ready
+        console.log('update via API');
+      } else {
+        followsYou = followUtils.followsYou(this.model.id);
+      }
+      // if this page has followed the viewer add them
+      if (followsYou) {
+        this.followCol.add({ guid: app.profile.id });
+        // if this page has unfollowed the viewer remove them
+      } else if (!followsYou) {
+        this.followCol.remove(app.profile.id); // remove by id
+      }
+    }
+  }
+
+  className() {
+    return 'userPageFollow flexRow';
+  }
+
+  render() {
+    this.$el.empty();
+    // TODO: add pagination if the collection is Followers
+    if (this.followCol.length) {
+      this.followCol.forEach((follow) => {
+        const user = this.createChild(userShort, {
```

Comment:
ok, let's not forget that because as it is now we have a non-insignificant memory leak. It's a really quick one to fix and have off our heads, but I guess we could defer as long as we don't forget. And.. until then, any type of scale metrics we'd do on this guy are not so accurate.